src/open-uri.c

```diff
@@ -522,15 +532,14 @@ handle_open_in_thread_func (GTask *task,
   if (skip_app_chooser || (!always_ask && (latest_count >= latest_threshold)))
     {
       /* If a recommended choice is found, just use it and skip the chooser dialog */
-      launch_application_with_uri (skip_app_chooser ? choices[0] : latest_id,
-                                   uri,
-                                   parent_window,
-                                   writable);
-
+      gboolean result = launch_application_with_uri (skip_app_chooser ? choices[0] : latest_id,
+                                                     uri,
+                                                     parent_window,
+                                                     writable);
       if (request->exported)
         {
           g_variant_builder_init (&opts_builder, G_VARIANT_TYPE_VARDICT);
-          xdp_request_emit_response (XDP_REQUEST (request), 0, g_variant_builder_end (&opts_builder));
+          xdp_request_emit_response (XDP_REQUEST (request), result ? 0 : 2, g_variant_builder_end (&opts_builder));
```

Comment:
I assume the `2` here means ‘reject’ or similar? If so, it would be good to have this encoded in an enum (I realise that is a pre-existing problem, so don’t worry about fixing it if you don’t have time).

Reply:
Yeah, it's an existing problem. I'll try to fix it